accio-base/src/main/java/io/accio/base/sqlrewrite/analyzer/StatementAnalyzer.java

```diff
@@ -150,55 +161,25 @@ protected Scope visitTable(Table node, Optional<Scope> scope)
                 analysis.addModelNodeRef(NodeRef.of(node));
-                List<Field> fields = ImmutableList.of();
-                if (accioMDL.getModel(tableName.getSchemaTableName().getTableName()).isPresent()) {
-                    fields = accioMDL.getModel(tableName.getSchemaTableName().getTableName())
-                            .map(Model::getColumns)
-                            .orElseGet(ImmutableList::of)
-                            .stream()
-                            .map(column -> Field.builder()
-                                    .modelName(tableName)
-                                    .columnName(column.getName())
-                                    .name(column.getName())
-                                    .build())
-                            .collect(toImmutableList());
-                }
-                else if (accioMDL.getMetric(tableName.getSchemaTableName().getTableName()).isPresent()) {
-                    fields = accioMDL.getMetric(tableName.getSchemaTableName().getTableName())
-                            .map(Metric::getColumns)
-                            .orElseGet(ImmutableList::of)
-                            .stream()
-                            .map(column -> Field.builder()
-                                    .modelName(tableName)
-                                    .columnName(column.getName())
-                                    .name(column.getName())
-                                    .build())
-                            .collect(toImmutableList());
-                }
-                else if (accioMDL.getCumulativeMetric(tableName.getSchemaTableName().getTableName()).isPresent()) {
-                    CumulativeMetric cumulativeMetric = accioMDL.getCumulativeMetric(tableName.getSchemaTableName().getTableName()).get();
-                    fields = ImmutableList.of(
-                            Field.builder()
-                                    .modelName(tableName)
-                                    .columnName(cumulativeMetric.getWindow().getName())
-                                    .name(cumulativeMetric.getWindow().getName())
-                                    .build(),
-                            Field.builder()
-                                    .modelName(tableName)
-                                    .columnName(cumulativeMetric.getMeasure().getName())
-                                    .name(cumulativeMetric.getMeasure().getName())
-                                    .build());
+                List<Field> fields = collectFieldFromMDL(tableName);
+
+                // if catalog and schema matches, but table name doesn't match any model, we assume it's a remote data source table
+                if (fields.isEmpty()) {
+                    return Scope.builder()
+                            .parent(scope)
+                            .isDataSourceScope(true)
+                            .build();
```

Comment:
If I remember correctly, we don't allow user to directly select remote data source table, not sure why we need this.

Reply:
To analyze the type of an identifier or a dereference expression, we need to get the correct metadata from models. Refer to the pattern to generate the model query:
```sql
WITH Orders as (
SELECT
  o_orderkey as orderkey
  ...
FROM
   (
     SELECT * FROM bq.tpch.orders
   ) Orders
)
```
The query for the remote data source is always wrapped in an `AliasedRelation` and the alias name will be the model name. The scoping fields of this subquery is always empty because we can't get the column metadata from the remote data source. Thus, we give a tag `isDataSourceScope` for this kind of relation, then get the column metadata based on the alias name(model name) from MDL to build the correct scope.

Reply:
Thanks for explanation, I got mislead by the comment `we assume it's a remote data source table`, actually we still get metadata from mdl, not remote data source.